File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.go
func divideArray(nums []int) bool {
    freq := make(map[int]int)

    for _, num := range nums {
        freq[num]++
    }

    for _, val := range freq {
        if val%2 != 0 {
            return false
        }
    }

    return true
}

File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.rs
use std::collections::HashMap;

impl Solution {
    pub fn divide_array(nums: Vec<i32>) -> bool {
        let mut freq = HashMap::new();

        for num in nums {
            *freq.entry(num).or_insert(0) += 1;
        }

        for count in freq.values() {
            if count % 2 != 0 {
                return false
            }
        }

        true
    }
}